### ISSUE TITLE
Restic backup fails after changing backup location

### ISSUE THREAD

**Issue by vitobotta** [vmware-tanzu/velero#2109]
**What steps did you take and what happened:**
I migrated my backup data from DigitalOcean Spaces to Exoscale Object Storage. The backups work fine apart from the Restic backups, which fail with `invalid id "a476b9b4": no matching ID found`. The first time I just changed the s3 settings in the backup location, but I've also tried uninstalling Velero completely and reinstalling with the new settings, same error.

**What did you expect to happen:**
I would expect Restic to continue backing up to the new location.

**The output of the following commands will help us better understand what's going on**:

* `kubectl logs deployment/velero -n velero`
[https://privatebin.net/?aa037737da086754#2mbShce7o7Lr3epArNoJE3DQpebzuSCuaNpSLqdQVoEG](https://privatebin.net/?aa037737da086754#2mbShce7o7Lr3epArNoJE3DQpebzuSCuaNpSLqdQVoEG)

* `velero backup describe <backupname>` or `kubectl get backup/<backupname> -n velero -o yaml`
[https://privatebin.net/?f0f842793e25bffc#ECcVMJLFqgo71EDwcppDHQwbVsJyayRVt2vgE4p3mtvu](https://privatebin.net/?f0f842793e25bffc#ECcVMJLFqgo71EDwcppDHQwbVsJyayRVt2vgE4p3mtvu)
[https://privatebin.net/?879d856ac3e4899e#5hj55e34NgwjFpfXNQTn9jL3kfYZMYd6LYYbjn13c7Xd](https://privatebin.net/?879d856ac3e4899e#5hj55e34NgwjFpfXNQTn9jL3kfYZMYd6LYYbjn13c7Xd)

* `velero backup logs <backupname>`
[https://privatebin.net/?8593023f185b18c0#7jcCVdAZKpSoJG9JAKF7pLDUZyAniTd8PNf5n3Wv5mTT](https://privatebin.net/?8593023f185b18c0#7jcCVdAZKpSoJG9JAKF7pLDUZyAniTd8PNf5n3Wv5mTT)

**Anything else you would like to add:**
I thought there might be a problem with the Restic repository but I tried connecting to it from my Mac and could run check/snapshots commands without any problems.

**Environment:**

- Velero version (use `velero version`): 
```
Client:
	Version: v1.2.0
	Git commit: -
Server:
	Version: v1.2.0
```

- Velero features (use `velero client config get features`): 
```
features: <NOT SET>
```

- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.2", GitCommit:"c97fe5036ef3df2967d086711e6c0c405941e14b", GitTreeState:"clean", BuildDate:"2019-10-15T23:42:50Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.3", GitCommit:"b3cbbae08ec52a7fc73d334838e18d17e8512749", GitTreeState:"clean", BuildDate:"2019-11-13T11:13:49Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}
```

- Kubernetes installer & version:
Rancher 2.3.3

- Cloud provider or hardware configuration:
4 cores, 8GB RAM, 200GB storage on each node

- OS (e.g. from `/etc/os-release`):
Ubuntu 18.04.3 LTS

**Comment by vitobotta** [comment]
Uhm I think I know what happened. 

When I updated the backup location for the new provider Exoscale, I did a backup to test BUT I didn't update the restic repo resource as well, so that was still pointing out to the bucket in DigitalOcean. I didn't restart the restic pods either, so I guess that test backup succeeded but put the Velero data into Exoscale and the restic data into DigitalOcean, completing successfully...

After deleting the DigitalOcean bucket, but still without updating repo resource nor restarting the restic pods, the daily backup failed this morning; I am not sure why, but even after updating the restic repo resource with the new restic identifier and restarting the restic pods, the subsequent backups failed.

Reinstalling Velero didn't help, because there seems to be some connection between the restic snapshots I guess because of the incremental backups? Can you confirm?

Anyway I deleted the failed backups AND the last backup that succeeded but split the data into the two different providers, and then I took a couple of backup which completed successfully, including restic volumes.

It would be nice if the documentation clarified a little what to do to change a backup location and what can go wrong.

**Comment by a-mccarthy** [comment]
To summarize, when updating the backup location, 
1. Update the location in Velero 
2. Update the Rest repo location
3.  restart the Rest pods

Are those all the steps we should clarify?

**Comment by vitobotta** [comment]
@a-mccarthy I think so, thanks :)

**Comment by eleanor-millman** [comment]
@ashish-amarnath @zubron Nolan thinks you two might have solved this in 1.6. Can you let us know?
